### PR TITLE
[TASK] Deprecate Binding context menu item to `this`

### PR DESCRIPTION
**Pull request by linawolf**
resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/155
releases: main

### CODE REVIEW

File: Documentation/ApiOverview/Backend/ContextualMenu.rst [hunk 1]
```diff
@@ -312,3 +312,22 @@ If :yaml:`autoconfigure` is disabled you can manually register a context menu it
 by adding the tag :yaml:`backend.contextmenu.itemprovider`:
 
 ..  include:: /CodeSnippets/Tutorials/ContextMenu/ManualServicesYaml.rst.txt
+
+Migration of the JavaScript
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/ContextualMenu.rst:316]:
```suggestion
Migration of JavaScript
```

Hmm ... don't like the header, find it that better: "Binding this to context menu" -> it makes clearer what is the purpose and search results may get better.

File: Documentation/ApiOverview/Backend/ContextualMenu.rst [hunk 2]
```diff
@@ -312,3 +312,22 @@ If :yaml:`autoconfigure` is disabled you can manually register a context menu it
...
+    assigned callback actions is adapted to pass the :js:`dataset` as the
+    third argument.
+
+    Binding the context menu item to :js:`this` in the callback is marked as
+    deprecated.
+
+To access data attributes, use the :js:`dataset` argument passed as the third
+argument in the context menu callback action.
+
+.. literalinclude:: _dataset-version-11.js
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/ContextualMenu.rst:332]: The snippet is not rendered, first it is in "_ContextMenu/_dataset-version-11.js" . but changing does not help, maybe the ":diff:" thingy?

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Backend/ContextualMenu.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ApiOverview/Backend/ContextualMenu.rst]: It works with the correct paths, but the result is not so nice, so I changed it to two different code-snippets

File: Documentation/ApiOverview/Backend/ContextualMenu.rst [hunk 3]
```diff
@@ -312,3 +312,25 @@ If :yaml:`autoconfigure` is disabled you can manually register a context menu it
...
+    Binding the context menu item to :js:`this` in the callback is marked as
+    deprecated.
+
+To access data attributes, use the :js:`dataset` argument passed as the third
+argument in the context menu callback action.
+
+.. literalinclude:: _ContextMenu/_dataset-version-11.js
+   :caption: Before migration (TYPO3 v11)
+
+.. literalinclude:: _ContextMenu/_dataset-version-11.js
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/ContextualMenu.rst:335]:
```suggestion
.. literalinclude:: _ContextMenu/_dataset.js
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**